shared/fs/container.js

```diff
@@ -1,16 +1,57 @@
 // @flow
+import * as I from 'immutable'
 import {connect, type TypedState, type Dispatch} from '../util/container'
 import * as FSGen from '../actions/fs-gen'
-import Fs from '.'
+import Files from '.'
+import {navigateAppend, navigateUp} from '../actions/route-tree'
+import * as Types from '../constants/types/fs'
 
-const mapStateToProps = (state: TypedState) => ({
-  counter: state.fs.counter,
-  you: state.config.username,
+type StateProps = {
+  you: ?string,
+  path: Types.Path,
+  items: I.List<string>,
+  pathItems: I.Map<Path, PathItem>,
+}
+
+type DispatchProps = {
+  onViewFolder: (path: Types.Path) => void,
+}
+
+const mapStateToProps = (state: TypedState, ownProps): StateProps => {
+  var path = state.fs.defaultPath
+  if (ownProps.routeProps) {
+    path = ownProps.routeProps.get('path', path)
+  }
+  const items = state.fs.pathItems.get(path).children || I.List()
+  return {
+    you: state.config.username,
+    path: path,
+    items: items,
+    pathItems: state.fs.pathItems,
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
+  onViewFolder: (path: Types.Path) => dispatch(navigateAppend([{props: {path}, selected: 'folder'}])),
+  onViewFile: (path: Types.Path) => console.log("Cannot view files yet."),
+  onBack: () => dispatch(navigateUp()),
 })
 
-const mapDispatchToProps = (dispatch: Dispatch) => ({
-  increase: () => dispatch(FSGen.createIncreaseCount({amount: 1})),
-  increase10: () => dispatch(FSGen.createIncreaseCount({amount: 10})),
+const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps) => ({
+  name: Types.getPathName(stateProps.path),
+  path: stateProps.path,
+  visibility: Types.getPathVisibility(stateProps.path),
+  items: stateProps.items.map(name => {
```

Comment:
it may make sense for this to actually be the representation of children in the tree. we should discuss